Apps/Admin/Server/Services/AgentAccessService.cs

```diff
@@ -95,11 +96,11 @@ public async Task<AdminAgent> ProvisionAgentAccessAsync(AdminAgent agent, Cancel
             await this.keycloakAdminApi.AddUserRolesAsync(createdUser.UserId.GetValueOrDefault(), roles, jwtModel.AccessToken, ct);
 
             string[] splitString = createdUser.Username.Split('@');
-            if (splitString.Length != 2)
-            {
-                string errorMessage = $"Agent username format is invalid: {createdUser.Username}";
-                throw new ProblemDetailsException(ExceptionUtility.CreateProblemDetails(errorMessage, HttpStatusCode.InternalServerError, nameof(AgentAccessService)));
-            }
+            InlineValidator<string[]> validator = new();
+            validator.RuleFor(s => s)
+                .Must(s => s.Length == 2)
+                .WithMessage($"Agent username format is invalid: {createdUser.Username}");
+            await validator.ValidateAndThrowAsync(splitString, ct);
```

Comment:
This is  a good example, the validation exception here now produces a 400 and not a 500, the structure of the username is not for the user input to control so it would be incorrect to say the fault lies from the client side.